conan/internal/model/recipe_ref.py

```diff
@@ -1,8 +1,13 @@
 from conan.api.model import RecipeReference
+from conan.api.output import ConanOutput
 
 
 def ref_matches(ref, pattern, is_consumer):
     if not ref or not str(ref):
         assert is_consumer
         ref = RecipeReference.loads("*/*")  # FIXME: ugly
+    if "[" in pattern or "]" in pattern:
```

Comment:
Can we check just first and last? Or even only first?

```suggestion
    if pattern.startswith("["):
```